src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPathExtensions.cs

```diff
@@ -52,16 +62,86 @@ public static IEdmNavigationSource NavigationSource(this ODataPath path)
         /// <returns>True if the resource if a resource set or collection of primitive or complex types. False otherwise.</returns>
         public static bool IsCollection(this ODataPath path)
         {
+            if (path == null)
+            {
+                throw Error.ArgumentNull(nameof(path));
+            }
+
             return path.LastSegment.TranslateWith(new IsCollectionTranslator());

```

Comment:
```suggestion
            return path.LastSegment?.TranslateWith(IsCollectionTranslator.Instance);
```

Please introduce a singleton instance of `IsCollectionTranslator` and use that there to reduce the number of object allocations

Reply:
#Resolved